### PR TITLE
WT-9847 Ensure that the truncation commit has resolved before allowing reconciliation to consider it

### PR DESCRIPTION
**Pull request by luke-pearson**

### CODE REVIEW

File: src/reconcile/rec_child.c [hunk 1]
```diff
@@ -35,13 +36,31 @@ __rec_child_deleted(
...
+     *     its prepare state set to resolved. At this stage the truncate can appear to be visible.
+     *   - After the operations have been resolved the page delete structure is marked as being
+     *     committed.
+     *
+     * Given the order of these operations we must perform the inverse sequence. First check the
+     * committed flag and then check the visibility. There is a concurrency concern here as if the
+     * write to the page delete structure is reordered by the compiler we may see it be set early.
+     * However this is handled by locking the ref in the commit path. Additionally this function
+     * locks the ref. Thus so setting the page delete structure committed flag cannot overlap with
+     * us checking the flag.
```
Comment (korteland) [inline src/reconcile/rec_child.c:55]: Couple of nitpicks: a comment in the first paragraph ("the visiblity, which") would help with readability.

The CPU can also reorder writes, so I'd probably just say "is reordered".

And "Thus so" reads a little weird, maybe pick one?

**Review by korteland — COMMENTED** [review]
A couple of nitpicks, plus I think we should include the test/format failure config since that's a more reliable local repro.

File: src/reconcile/rec_child.c (hunk 1 above)
Comment (luke-pearson) [reply src/reconcile/rec_child.c]: [5b8a7f7](https://github.com/wiredtiger/wiredtiger/pull/8693/commits/5b8a7f7d9ddfea232f8bb36d6f27210344d8e501)

**Review by korteland — APPROVED** [review]
Thanks! LGTM

**Review by agorrod — APPROVED** [review]
LGTM - great work. The comment is good, and I'm confident in the change. It's been a long path, I appreciate you (both Will and Luke) finding a way through.

**Review by kommiharibabu — APPROVED** [review]
LGTM

**Comment by luke-pearson** [comment]
evergreen merge

**Pull request merged**